dataverse-persistence/src/main/java/edu/harvard/iq/dataverse/persistence/datafile/ingest/IngestReport.java

```diff
@@ -121,6 +130,30 @@ public void setEndTime(Date endTime) {
         this.endTime = endTime;
     }
 
+    public List<String> getReportArguments() {
+        return reportArguments;
+    }
+
+    // -------------------- LOGIC --------------------
+
+    public String getIngestReportMessage() {
+        IngestError errorKey = getErrorKey();
+        if (errorKey != null) {
+
+            if (!getReportArguments().isEmpty()) {
+                String errorMessage = IngestError.getErrorMessage(errorKey, getReportArguments());
```

Comment:
Why not use the filed directly instead of via getter here?

Reply:
Well I would have to duplicate the internals of the method, I just didn't see a reason to.

Reply:
I meant `reportArguments` instead of `getReportArguments()` - there's no internals there to speak of ;)

Reply:
Ohh, there is no real reason, is there a reason you prefer direct field call instead of calling getter? You think it's more readable (One thing I can think of, is that in getter there could be some kind of logic, if you call it directly you are sure there won't be)?

Reply:
The reason is encapsulation. Getters are meant to expose object internal state to the outside. Additionally non final getters can be affected by inheritance. In general they are a tool to break encapsulation. Meanwhile I'd like to keep the object method implementation well encapsulated, operate on internal object state directly and keep it closed to modification. Sure, there are use cases to use methods to get internal state instead, but that's rather when you expect sub-classes to contribute, not when it's just a plain getter.